substrate/network-libp2p/src/network_state.rs

```diff
@@ -180,8 +181,21 @@ impl NetworkState {
 				PeersStorage::Json(peerstore)
 			} else {
 				warn!(target: "sub-libp2p", "Failed to open peer storage {:?} \
-					; peers won't be saved", path);
-				PeersStorage::Memory(MemoryPeerstore::empty())
+					; peers file will be reset", path);
+				fs::remove_file(&path).unwrap();
+				while Path::new(&path).exists() {
```

Comment:
is this loop really necessary?

Reply:
`fs::remove_file` may, or not, be performed synchronously, this seems to be OS dependant. On my machine (OSX), this loop is useless. It may not be the case for other systems.

Reply:
I'm a bit afraid that this will deadlock if the user's system decides to do some black magic, but maybe I'm not rational.

Reply:
I find that paranoically-healthy @tomaka :)

I guess I could add some sort of timeout. I am not sure which OS we target but an option is also to check that it works for OSX, Linux and Windows and remove the loop entirely.

Your take on that one, just tell me.